packages/liquidator/src/liquidationStrategy.js

```diff
@@ -142,46 +129,44 @@ module.exports = (
-          position: position,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          syntheticTokenBalance: syntheticTokenBalance.toString(),
-          maxTokensToLiquidateWei: maxTokensToLiquidateWei ? maxTokensToLiquidateWei.toString() : null,
-          whaleDefenseFundWei: whaleDefenseFundWei.toString(),
-          currentBlockTime,
-          ...logInfo
-        });
-        // this assume empMinSponsorSize will always be <= maxTokensToLiquidate
-        return createLiquidationParams({
-          sponsor: position.sponsor,
-          // Only liquidating the minimum size to extend withdraw
-          tokensToLiquidate: minSponsorSize,
-          currentBlockTime,
-          maxCollateralPerToken
-        });
-      }
-    }
+    let tokensToLiquidate = toBN(0);
 
-    // WDF strategy not activated, calculate tokens to liquidate respecting all constraints
-    const tokensToLiquidate = calculateTokensToLiquidate({
-      syntheticTokenBalance,
-      positionTokens: position.numTokens,
-      maxTokensToLiquidateWei,
-      withdrawRequestPending: hasWithdrawRequestPending({ position, currentBlockTime })
-    });
+    if (!wdfActive || !hasWithdrawRequestPending({ position, currentBlockTime })) {
+      // 1) If either WDF is not activated or there is no withdraw pending, then
+      //    submit liquidation using all capital
+      //     - This will include withdraws that have expired already.
+      tokensToLiquidate = calculateTokensToLiquidate({
+        syntheticTokenBalance,
+        positionTokens: position.numTokens,
+        maxTokensToLiquidateWei
+      });
+    } else if (canLiquidateFully({ position, syntheticTokenBalance })) {
+      // 2) If you can fully liquidate, do it, regardless of whether WDF is activated.
+      tokensToLiquidate = calculateTokensToLiquidate({
+        syntheticTokenBalance,
+        positionTokens: position.numTokens,
+        maxTokensToLiquidateWei
+      });
+    } else if (
+      wdfActive &&
+      hasWithdrawRequestPending({ position, currentBlockTime }) &&
+      passedDefenseActivationPercent({ position, currentBlockTime }) &&
+      canLiquidateMinimum({ position, syntheticTokenBalance })
+    ) {
+      // 3) If WDF is active and withdraw has passed liveness, then submit min liquidation
+      emit("info", {
+        message: "Liquidator bot is extending withdraw deadline, funds may need to be added to liquidate full position",
+        position: position,
+        maxLiquidationPrice: maxCollateralPerToken.toString(),
+        syntheticTokenBalance: syntheticTokenBalance.toString(),
+        maxTokensToLiquidateWei: maxTokensToLiquidateWei ? maxTokensToLiquidateWei.toString() : null,
+        currentBlockTime,
+        ...logInfo
+      });
+      tokensToLiquidate = minSponsorSize;
+    } else {
+      // 4) Do nothing.
+    }
 
     // check if we should try to liquidate this position
     if (shouldLiquidate({ tokensToLiquidate })) {
```

Comment:
i wonder if this function is still needed after all the checks in the if/else?

Reply:
Currently the function checks that `tokensToLiquidate >= 0`, but I'm not sure that's even possible now. I can remove if you feel OK with that.